gremlin-client/src/structure/map.rs

```diff
@@ -77,12 +77,37 @@ impl Map {
         self.0.iter()
     }
 
+    pub fn into_iter(self) -> IntoIter<GKey, GValue> {
+        self.0.into_iter()
+    }
+
     ///Returns a reference to the value corresponding to the key.
     pub fn get<T>(&self, key: T) -> Option<&GValue>
     where
         T: Into<GKey>,
     {
-        self.0.get(&key.into())
+        let key = key.into();
+        self.0
+            .get(&key)
+            .or_else(|| self.backwards_compatability_get(key))
+    }
+
+    fn backwards_compatability_get(&self, key: GKey) -> Option<&GValue> {
+        //The GraphBinary protocol may be returning properties
+        //as the distinct "T" type (T::Id) whereas older protocols
+        //would have just sent them as GKey::String("id"), etc
+        //This function is intended as a fallback for backwards compatability that if a caller
+        //requests a get or try_get for "id" that we also then attempt it T::Id or one of its siblings
+        //This also maintains the representation needed for the derive crate to find
+        //the types since the derive crate can't discriminate between a vertex property called "id"
+        //or a T::Id since they're both just called "id"
+        match key {
+            GKey::String(string) => match TryInto::<T>::try_into(string.as_str()) {
+                Ok(fallback_key) => self.0.get(&fallback_key.into()),
+                Err(_) => None,
+            },
+            _ => None,
+        }
```

Comment:
@wolf4ood really wanted to call attention to this. Originally I was intended to just make users make the jump to calling a map.get(T::Id) vs map.get("id") but once I fixed the derive tests I hit a wall that the derive macro logic can't make that jump (I think?).

Namely the derive macro can't really know if the final struct should be using a property called "id" or "T::id" now that GraphBinary discriminates between the two. So I was hesitant to "fix" the problem on that side.

Doing it here means anyone who was doing `map.get("id")` will transparently get switched to `map.get(T::id)` under the covers if their initial get attempt returns a `None` now. That way it's only a perf hit as a fallback, whereas the derive macro would have had to do that for every property I think.

Anyways, wanted to call this out because it's kind of a catch 22 situation that we're forced into I think because of the derive macros.